tests/test_client.py

```diff
@@ -620,109 +621,118 @@ def test_quota_overview(client: GGClient):
-        "status": "active",
-        "triggered_at": "2019-08-22T14:15:22Z",
-        "revoked_at": None,
-        "open_events_count": 2,
-        "type": "AWS",
-        "creator_id": 122,
-        "revoker_id": None,
-        "creator_api_token_id": None,
-        "revoker_api_token_id": None,
-        "token": {"access_token_id": "AAAA", "secret_key": "BBB"},
-        "tags": ["publicly_exposed"],
-    }
-
-    request_mock.return_value = mock_response
+    mock_response = responses.post(
+        url=client._url_from_endpoint("honeytokens", "v1"),
+        content_type="application/json",
+        status=201,
+        json={
+            "id": "d45a123f-b15d-4fea-abf6-ff2a8479de5b",
+            "name": "honeytoken A",
+            "description": "honeytoken used in the repository AA",
+            "created_at": "2019-08-22T14:15:22Z",
+            "gitguardian_url": "https://dashboard.gitguardian.com/workspace/1/honeytokens/d45a123f-b15d-4fea-abf6-ff2a8479de5b",  # noqa: E501
+            "status": "active",
+            "triggered_at": "2019-08-22T14:15:22Z",
+            "revoked_at": None,
+            "open_events_count": 2,
+            "type": "AWS",
+            "creator_id": 122,
+            "revoker_id": None,
+            "creator_api_token_id": None,
+            "revoker_api_token_id": None,
+            "token": {"access_token_id": "AAAA", "secret_key": "BBB"},
+            "tags": ["publicly_exposed"],
+        },
+    )
 
     result = client.create_honeytoken(
         name="honeytoken A",
         description="honeytoken used in the repository AA",
         type_="AWS",
     )
 
-    assert request_mock.called
+    assert mock_response.call_count == 1
     assert isinstance(result, HoneytokenResponse)
 
 
-@patch("requests.Session.request")
+@responses.activate
 def test_create_honeytoken_error(
-    request_mock: Mock,
     client: GGClient,
 ):
     """
     GIVEN a ggclient
     WHEN calling create_honeytoken with parameters without the right access
-    THEN I get a Detail objects containing the error detail
+    THEN I get a Detail object containing the error detail
     """
-    mock_response = Mock(spec=Response)
-    mock_response.headers = {"content-type": "application/json"}
-    mock_response.status_code = 400
-    mock_response.json.return_value = {
-        "detail": "Not authorized",
-    }
-
-    request_mock.return_value = mock_response
+    mock_response = responses.post(
+        url=client._url_from_endpoint("honeytokens", "v1"),
+        content_type="application/json",
+        status=400,
+        json={
+            "detail": "Not authorized",
+        },
+    )
 
     result = client.create_honeytoken(
         name="honeytoken A",
         description="honeytoken used in the repository AA",
         type_="AWS",
     )
 
-    assert request_mock.called
+    assert mock_response.call_count == 1
     assert isinstance(result, Detail)
-    result.status_code == 400
```

Comment:
Shouldn't we better add the missing `assert` rather than deleting this line ?

Reply:
I think that's not the job of this test: checking that the API returns 400 when it receives an invalid request is the job of the API test. In this test what we want to verify is how the client behaves when it receives a 400 response.